### PR TITLE
test(NODE-3494): add observeSensitiveCommands spec tests

### PR DESCRIPTION
**Pull request by alenakhineika**
#### What is changing?
- Sync with the latest [observeSensitiveCommands.json](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/tests/valid-pass/observeSensitiveCommands.json) and [observeSensitiveCommands.yml](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/tests/valid-pass/observeSensitiveCommands.yml) spec tests.
- Update the unified test runner to filter out sensitive hello and legacy hello commands with speculativeAuthenticate to be observed.

##### Is there new documentation needed for these changes?
None

#### What is the motivation for this change?
[NODE-3494](https://jira.mongodb.org/browse/NODE-3494)

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/tools/unified-spec-runner/entities.ts [hunk 1]
```diff
@@ -266,7 +288,10 @@ export class UnifiedMongoClient extends MongoClient {
 
   // NOTE: pushCommandEvent must be an arrow function
   pushCommandEvent: (e: CommandEvent) => void = e => {
-    if (!this.isIgnored(e)) {
+    if (
+      (this.observeSensitiveCommands === true || !this.isSensitiveCommand(e)) &&
```
Comment (aditi-khare-mongoDB) [inline test/tools/unified-spec-runner/entities.ts:292]: If `observeSensitiveCommands` is off, and a sensitive command is located, why not push it into `IgnoredEvents`? There is precedent for this in the [unified spec runner](https://github.com/mongodb/node-mongodb-native/blob/9126b3ef38af1ba06bd7f7cbd7061144417a6d7b/test/tools/unified-spec-runner/entities.ts#L218C5-L220C6)

File: test/tools/unified-spec-runner/entities.ts [hunk 2]
```diff
@@ -119,6 +119,7 @@ export class UnifiedMongoClient extends MongoClient {
   logCollector: { buffer: LogMessage[]; write: (log: Log) => void };
 
   ignoredEvents: string[];
+  observeSensitiveCommands = false;
```
Comment (aditi-khare-mongoDB) [inline test/tools/unified-spec-runner/entities.ts:122]: Is this line a duplication of the `false` in line 223, or is it necessary?

**Review by aditi-khare-mongoDB — CHANGES_REQUESTED** [review]

File: test/tools/unified-spec-runner/entities.ts (hunk 1 above)
Comment (alenakhineika) [reply test/tools/unified-spec-runner/entities.ts:303]: To locate a sensitive command, we need to check if the `command` or `reply` properties of the command event an empty object (according to [spec](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst) and [other drivers implementation](https://github.com/mongodb/mongo-rust-driver/commit/114a519ac2fc2bc2bb8ff3cf765b029919d2c5cd#diff-fd8356adaae0c4a7d9d11585800aaf2ec0513c67795c2b66ec4bc3764b22885eR90-R91)):

"Note that the command and replies for such events will already have been redacted by the driver. If false or not specified, events for commands containing sensitive information MUST be ignored."

We don't have this info yet in the UnifiedMongoClient constructor. When `observeSensitiveCommands` is off, the hello can have or hasn't data in reply.

Comment (aditi-khare-mongoDB) [reply test/tools/unified-spec-runner/entities.ts:292]: Update: Talked about this comment thread over zoom; IgnoredEvents is about ignoring event types not individual events.

**Review by aditi-khare-mongoDB — APPROVED** [review]

**Review by nbbeeken — APPROVED** [review]

**Comment by aditi-khare-mongoDB** [comment]
Failing tests are flaky tests reported in [NODE-5788](https://jira.mongodb.org/browse/NODE-5788)

**Pull request merged**